dask_expr/tests/test_shuffle.py

```diff
@@ -114,3 +114,12 @@ def test_shuffle_column_projection():
     df2 = df.shuffle("x")[["x"]].simplify()
 
     assert "y" not in df2.expr.operands[0].columns
+
+
+def test_shuffle_reductions():
+    pdf = pd.DataFrame({"x": list(range(20)) * 5, "y": range(100)})
+    df = from_pandas(pdf, npartitions=10)
+
+    assert df.shuffle("x").sum().optimize()._name == df.sum()._name  # This passes
+
+    assert df.shuffle("x").y.sum().optimize()._name == df.y.sum()._name  # This fails
```

Comment:
We can't support this right now 😢

Reply:
That would solve this problem, but it doesn't feel ideal.  I think that the underlying issue here is that we can't write optimizations that look up more than one expression.  I think I'm going to just fail to do this well today for now and leave it as future work.

Reply:
Yeah, I'm fine with this not performing great for a while in a way that I think we weren't ok with IO not performing great.